File: iOS/DittoPOS/Data/Payment.swift
//
//  Payment.swift
//  DittoPOS
//
//  Copyright © 2026 DittoLive Incorporated. All rights reserved.
//

import Foundation

enum PaymentType: String, Codable, CaseIterable {
    case cash, credit, debit, refund
}

enum PaymentStatus: String, Codable, CaseIterable {
    case incomplete, inProcess, complete, failed
}

struct Payment: Codable, Hashable, Equatable {
    let type: PaymentType
    let amount: Price
    let status: PaymentStatus
    let createdAt: Date

    init(
        type: PaymentType,
        amount: Price,
        status: PaymentStatus = .complete,
        createdAt: Date = Date()
    ) {
        self.type = type
        self.amount = amount
        self.status = status
        self.createdAt = createdAt
    }

    static func newPaymentId() -> String { UUID().uuidString }
}

File: Android/app/src/main/java/live/ditto/pos/core/data/Payment.kt
package live.ditto.pos.core.data

import kotlinx.datetime.Clock
import kotlinx.datetime.Instant
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
enum class PaymentType {
    @SerialName("cash")
    CASH,

    @SerialName("credit")
    CREDIT,

    @SerialName("debit")
    DEBIT,

    @SerialName("refund")
    REFUND
}

@Serializable
enum class PaymentStatus {
    @SerialName("incomplete")
    INCOMPLETE,

    @SerialName("inProcess")
    IN_PROCESS,

    @SerialName("complete")
    COMPLETE,

    @SerialName("failed")
    FAILED
}

@Serializable
data class Payment(
    val type: PaymentType,
    val amount: Price,
    val status: PaymentStatus = PaymentStatus.COMPLETE,
    @Serializable(with = DittoInstantSerializer::class)
    val createdAt: Instant = Clock.System.now()
) {
    companion object {
        fun newPaymentId(): String = UUID.randomUUID().toString()
    }
}
